pyramid/config/views.py

```diff
@@ -568,9 +568,8 @@ def add(self, view, order, accept=None, phash=None):
             else:
                 subset.append((order, view, phash))
                 subset.sort(key=operator.itemgetter(0))
-            accepts = set(self.accepts)
-            accepts.add(accept)
-            self.accepts = list(accepts) # dedupe
```

Comment:
Then a relevant sort function can be passed in. The use of set() already indicates that order does not matter but you have an edge case where two different views can match the same request in some circumstances (missing accepts-header).